sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
```

Comment:
Interesting touch to using a network analysis library for creating the SQL graphs. A few questions:

1. How big is this library? 
2. Are we doing complex enough DAG creation to require this library?

I don't have strong thoughts either way, just want to make sure we're asking the question. Certainly seems to make the code cleaner :).

Reply:
Yes, totally fair to ask this question.

I personally dont trust my graph algorithm skills enough to reimplement this from scratch. 😅 I am afraid, I am missing edge cases etc. That is why I decided to use _networkx_. It has builtin functions for cycle detection and finding the right order to execute statements in using dfs.